test/e2e/gardener/shoot/internal/rotation/shoot_access.go

```diff
@@ -80,157 +86,177 @@ func (v *ShootAccessVerifier) ExpectPreparingWithoutWorkersRolloutStatus(_ Gomeg
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Use dynamic ServiceAccount token from after preparation to access shoot", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			g.Expect(v.clientsPrepared.serviceAccountDynamic.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Use static ServiceAccount token from after preparation to access shoot", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			g.Expect(v.clientsPrepared.serviceAccountStatic.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Use admin kubeconfig with new CA to access shoot", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			shootClient, err := access.CreateShootClientFromAdminKubeconfig(ctx, v.GardenClientSet, v.Shoot)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(shootClient.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
+
+			v.clientsAfter.adminKubeconfig = shootClient
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Request new client certificate and using it to access shoot", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			shootClient, err := access.CreateTargetClientFromCSR(ctx, v.clientsAfter.adminKubeconfig, "e2e-rotate-csr-after")
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(shootClient.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
+
+			v.clientsAfter.clientCert = shootClient
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Request new dynamic token for a ServiceAccount and using it to access shoot", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			shootClient, err := access.CreateTargetClientFromDynamicServiceAccountToken(ctx, v.clientsAfter.adminKubeconfig, "e2e-rotate-sa-dynamic-after")
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(shootClient.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
+
+			v.clientsAfter.serviceAccountDynamic = shootClient
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
+
+	It("Request new static token for a ServiceAccount and using it to access shoot", func(ctx SpecContext) {
+		Eventually(ctx, func(g Gomega) {
+			shootClient, err := access.CreateShootClientFromStaticServiceAccountToken(ctx, v.clientsAfter.adminKubeconfig, "e2e-rotate-sa-static-after")
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(shootClient.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
+
+			v.clientsAfter.serviceAccountStatic = shootClient
+		}).Should(Succeed())
+	}, SpecTimeout(time.Minute))
 }
 
 // Cleanup is passed to ginkgo.DeferCleanup.
-func (v *ShootAccessVerifier) Cleanup(ctx context.Context) {
-	if v.Config.GardenerConfig.ExistingShootName == "" {
+func (v *ShootAccessVerifier) Cleanup() {
+	if flag.Lookup("existing-shoot-name").Value.String() == "" {
```

Comment:
Can you add a `TODO` similar to https://github.com/gardener/gardener/blob/8b123e4119b88809df55b0d1bd0739f8a2cef546/test/e2e/gardener/shoot/internal/shoot.go#L44-L45?
Eventually, this should use the newly defined flag in the `e2e` package.